src/SprykerSdk/Zed/SprykGui/Communication/Controller/AbstractController.php

```diff
@@ -18,23 +17,15 @@
 class AbstractController extends SprykerAbstractController
 {
     /**
+     * @throws \Symfony\Component\HttpKernel\Exception\NotFoundHttpException
+     *
      * @return void
      */
     public function initialize(): void
     {
         parent::initialize();
 
-        $this->assertNonProductionEnvironment();
-    }
-
-    /**
-     * @throws \Symfony\Component\HttpKernel\Exception\NotFoundHttpException
-     *
-     * @return void
-     */
-    protected function assertNonProductionEnvironment(): void
-    {
-        $isProductionEnvironment = Environment::isProduction();
+        $isProductionEnvironment = APPLICATION_ENV === 'production';
         $isCli = PHP_SAPI === 'cli';
 
         if (!$isProductionEnvironment || $isCli) {
```

Comment:
I would like to remove `isProductionEnvironment` condition we cannot determinate environment type from this scope. Production environment must not contain this module included at all.

@dereuromark Please confirm you agree or disagree. Thanks

Reply:
But how do you in the future ensure the prod vs development switch? In a clean and safe way?

Reply:
I suppose strict recommendation to use `--no-dev` mode for production described in Spryker docs would be enough.